src/lib.rs

```diff
@@ -82,6 +84,7 @@ impl Default for Config {
             receive_window: DEFAULT_CREDIT,
             max_buffer_size: 1024 * 1024,
             max_num_streams: 8192,
+            max_pending_frames: 2048,
```

Comment:
It is probably difficult to come up with criteria for a good default value, but how about aligning it with `max_num_streams` by the simple reasoning that there should be on average at least 1 free slot for an outgoing frame per stream, i.e. the max number of streams loosely defines the "bandwidth" in terms of the number of frames sent at a time for a healthy connection?

Reply:
It is indeed not easy to find a good default. The thing to keep in mind is the upper bound of memory usage we want to allow.  This would be `max_pending_frames * receive_window` (of the remote and ignoring 12 bytes for the header per frame). Using the proposed default settings this would amount to 512 MiB. With `max_pending_frames = 8192` it would go up to 2 GiB. I have no strong opinion which one to chose, but considering that processes may use more than one connection this could easily make them prime targets for an OOM killer. But maybe 512 MiB per connection would already be too much. I do not know.